core/contracts/financial-templates/implementation/demo/DepositBox.sol

```diff
@@ -0,0 +1,346 @@
+        FixedPoint.Unsigned rawCollateral;
+    }
+
+    // Maps addresses to their deposit boxes. Each address can have only one position.
+    mapping(address => DepositBoxData) public depositBoxes;
+
+    // Unique identifier for DVM price feed ticker.
+    bytes32 public priceIdentifier;
+
+    // Similar to the rawCollateral in DepositBoxData, this value should not be used directly.
+    // _getFeeAdjustedCollateral(), _addCollateral() and _removeCollateral() must be used to access and adjust.
+    FixedPoint.Unsigned public rawTotalDepositBoxCollateral;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event NewDepositBox(address indexed user);
+    event EndedDepositBox(address indexed user);
+    event Deposit(address indexed user, uint256 indexed collateralAmount);
+    event RequestWithdrawal(address indexed user, uint256 indexed collateralAmount, uint256 requestPassTimestamp);
+    event RequestWithdrawalExecuted(
+        address indexed user,
+        uint256 indexed collateralAmount,
+        uint256 exchangeRate,
+        uint256 requestPassTimestamp
+    );
+    event RequestWithdrawalCanceled(
+        address indexed user,
+        uint256 indexed collateralAmount,
+        uint256 requestPassTimestamp
+    );
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
+
+    modifier noPendingWithdrawal(address user) {
+        _depositBoxHasNoPendingWithdrawal(user);
+        _;
+    }
+
+    /**
+     * @notice Construct the DepositBox.
+     * @param _collateralAddress ERC20 token to be deposited.
+     * @param _finderAddress UMA protocol Finder used to discover other protocol contracts.
+     * @param _priceIdentifier registered in the DVM, used to price the ERC20 deposited.
+     * The price identifier consists of a "base" asset and a "quote" asset. The "base" asset corresponds to the collateral ERC20
+     * currency deposited into this account, and it is denominated in the "quote" asset on withdrawals.
+     * @param _timerAddress Contract that stores the current time in a testing environment.
+     * Must be set to 0x0 for production environments that use live time.
+     */
+    constructor(
+        address _collateralAddress,
+        address _finderAddress,
+        bytes32 _priceIdentifier,
+        address _timerAddress
+    )
+        public
+        ContractCreator(_finderAddress)
+        FeePayer(_collateralAddress, _finderAddress, _timerAddress)
+        nonReentrant()
+    {
+        require(_getIdentifierWhitelist().isIdentifierSupported(_priceIdentifier), "Unsupported price identifier");
+
+        priceIdentifier = _priceIdentifier;
+    }
+
+    /**
+     * @notice This should be called after construction of the DepositBox and handles registration with the Registry, which is required
```

Comment:
prehaps add more to this comment to explain why you cant do this call from the constructor?

Reply:
Good call. To be clear, it cannot be called from constructor because the address needs to hold the `ContractCreator` role prior to registering itself with the Registry, but its address is not discovered until post-deployment.